File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        HashMap<String, Integer> dict = new HashMap<>();
        for (int i=0; i<name.length; i++) {
            dict.put(name[i], yearning[i]);
        }
        
        int[] answer = new int[photo.length];
        for (int i=0; i<photo.length; i++) {
            String[] people = photo[i];
            for (String man : people) {
                answer[i] += dict.getOrDefault(man, 0);
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.kt
class Solution {
    fun solution(names: Array<String>, yearning: IntArray, photo: Array<Array<String>>): IntArray {
        val answer = mutableListOf<Int>()
        val scores = names.zip(yearning.toTypedArray()).toMap()

        photo.forEachIndexed { index, photos ->
            answer.add(0)
            photos.forEach { name ->
                answer[index] += scores[name] ?: 0
            }
        }

        return answer.toIntArray()
    }
}